### PR TITLE
Wait for postgres service to be available

### PR DESCRIPTION
**Pull request by sayanarijit**
- Init DB scripts should wait for the postgres server on slower
machines.
- Ignore IDEs and code editor folders in gitignore.

### CODE REVIEW

**Comment by sayanarijit** [comment]
Need to set the JWT_SECRET env var in circleci to pass.

File: .docker/devdb.sh [hunk 1]
```diff
@@ -1,13 +1,20 @@
 #!/bin/bash
 set -e
 
-psql "postgresql://$POSTGRES_USER@:5432/$POSTGRES_DB" -v ON_ERROR_STOP=1 <<-EOSQL
+until psql -U $POSTGRES_USER -c '\q'; do
+  >&2 echo "Postgres is unavailable - waiting..."
+  sleep 1
+done
```
Comment (zupo) [inline .docker/devdb.sh]: 👍

File: .docker/devdb.sh [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+
+psql -U $POSTGRES_USER -v ON_ERROR_STOP=1 <<-EOSQL
+  DROP DATABASE IF EXISTS conduit_dev;
+  DROP USER IF EXISTS conduit_dev;
   CREATE USER conduit_dev WITH PASSWORD '';
   CREATE DATABASE conduit_dev;
   GRANT ALL PRIVILEGES ON DATABASE conduit_dev TO conduit_dev;
 EOSQL
 
-psql -v ON_ERROR_STOP=1 "postgresql://$POSTGRES_USER@:5432/conduit_dev" <<-EOSQL
```
Comment (zupo) [inline .docker/devdb.sh]: I prefer the full URL for two reasons:

1. Explicit is better than implicit.
2. If I need to inspect the database with pgqwb/psql/whatever, I open up `Makefile`, copy the URL and paste it into console. Done. After this change, I would have to manually construct the URL.

File: etc/development.ini [hunk 3]
```diff
@@ -11,7 +11,8 @@ pyramid_force_https.structlog = false
 # We do not want force_https and heroku.client_addr from production.ini
 pyramid.includes = ""
 
-sqlalchemy.url = postgresql+psycopg2://conduit_dev@localhost/conduit_dev
+# Set DATABASE_URL env var to use a different database
+sqlalchemy.url = ${DATABASE_URL:-postgresql+psycopg2://conduit_dev@localhost/conduit_dev}
```
[the remark this answers is not in the document]
Comment (sayanarijit) [reply etc/development.ini]: With this we can easily run the app with a DB hosted on a different URL such as staging/reviewapps or production without needing to modifying the file.

[the remark this answers is not in the document]
Comment (sayanarijit) [reply etc/development.ini]: Other option is to import the dump locally which is better I guess. Right will remove it.

File: .docker/devdb.sh (hunk 2 above)
Comment (sayanarijit) [reply .docker/devdb.sh]: I remember I read this argument once before. Probably in this app... 😄

File: etc/development.ini (hunk 3 above)
[the remark this answers is not in the document]
Comment (zupo) [reply etc/development.ini]: I really don't trust env vars in development. They are so implitic. Sooner or later you will think you are using dev db, but miss that you are in that tab that has an env var override for production db, and booom.

**Comment by sayanarijit** [comment]
@zupo I think it's ready.

**Comment by zupo** [comment]
@sayanarijit: thanks for the poke, I hope to be able to look at it during the weekend

File: .docker/testdb.sh [hunk 4]
```diff
@@ -1,7 +1,14 @@
 #!/bin/bash
 set -e
 
-psql "postgresql://$POSTGRES_USER@:5432/$POSTGRES_DB" -v ON_ERROR_STOP=1 <<-EOSQL
```
Comment (zupo) [inline .docker/testdb.sh]: Why is `$POSTGRES_DB` part of URI no longer needed?

File: etc/production.ini [hunk 5]
```diff
@@ -21,8 +21,8 @@ pyramid_force_https.structlog = true
 
 sqlalchemy.url = ${DATABASE_URL}
 
-# secrets
-jwt.secret = ${JWT_SECRET}
+# secrets (must be set)
+jwt.secret = ${JWT_SECRET:?}
```
Comment (zupo) [inline etc/production.ini]: Why is this needed? How does it manifest?

Comment (sayanarijit) [reply etc/production.ini]: It will raise a runtime error if the variable is not defined. Same as bash.
```
bash-3.2$ echo ${UNSET_VAR:?}
bash: UNSET_VAR: parameter null or not set
bash-3.2$ echo $?
1
```

Comment (sayanarijit) [reply etc/production.ini]: It's also possible to use a custom error message:
```bash
bash-3.2$ echo ${UNSET_VAR:?custom error message}
bash: UNSET_VAR: custom error message
```

File: .docker/testdb.sh (hunk 4 above)
Comment (sayanarijit) [reply .docker/testdb.sh]: We are creating DB, users and granting privileges. Hence, we don't need to specify any database to connect to.
